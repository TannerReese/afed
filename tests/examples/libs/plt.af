# Test plotting functions

# Erroneous cases
plt.plot {jk: 5} [\x: x] = ``,
plt.plot {rows: 30, cols: 10} [4, 5] = ``,


{
plt.plot {
	ymin: -0.5, width: 3,
	rows: 30, cols: 80, divs: 3,
} [f, g] = ``,

	f x: 1 / (1 + x^2),
	g x: sin (2*x - 1),
},

plt.plot {
	xcenter: 3, width: 6,
	rows: 20, cols: 60, divs: 3,
} [\x: sin x / x] = ``,


