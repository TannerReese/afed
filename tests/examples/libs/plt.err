line 4, column 28 of "plt.af" Eval Error: Unknown options '["jk"]'
line 5, column 40 of "plt.af" Eval Error: Cannot call type number
2 Eval Errors encountered
