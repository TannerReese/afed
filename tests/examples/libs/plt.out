# Test plotting functions

# Erroneous cases
plt.Plot {jk: 5} [\x: x] = `Eval Error: Unknown options '["jk"]'`,
errored: 4 + plt.Plot {rows: 10, cols: 10},
errored.errors = `["Cannot call method arity on type number"]`,


{
plt.draw [f, g] $ [\x: 0.8, [0.5, 0.6], [-0.5, -0.1]] + p = `
             -1                         0                         1             
             |                          $                         |             
             |                          $                         |             
             |                          $                         |             
             |                          $                         |             
             |                          $                         |             
             |                          $                         |             
1.0000-------+---------------------,~~~~~~~~,---------------------+---,~~~~~~,--
             |                 ,~~~'    $   '~~~,                 | ,~'      '~~
             |              ,~~'        $       '~~,              ,~'           
~~~~~~~~~~~~~~~~~~~~~~~~~,~~'~~~~~~~~~~~~~~~~~~~~~~'~~,~~~~~~~~~,~'~~~~~~~~~~~~~
~,           |        ,~~'              $             '~~,     ,' |             
 ',          |     ,~~'                 $                '~~, ,'  |             
  ',         |  ,~~'                    $            O      ',',  |             
0.5',0-------,~~'-----------------------+-------------------,'-'~~,-------------
    ',   ,~~~'                          $                  ,'     '~~~,         
    ,',~~'   |                          $                 ,'      |   '~~~~,    
~~~~' ',     |                          $                ,'       |        '~~~~
       ',    |                          $               ,'        |             
        ',   |                          $              ,'         |             
         ',  |                          $             ,'          |             
          ', |                          $            ,'           |             
0==========!=+==========================#===========,'============+=============
           ',|                          $          ,'             |             
            ',            O             $          !              |             
             ',                         $         ,'              |             
             |',                        $        ,'               |             
             | ',                       $       ,'                |             
             |  ',                      $      ,'                 |             
             |   ',                     $     ,'                  |             
`,

	p: plt.Plot {
		ymin: -0.5, width: 3,
		rows: 30, cols: 80, divs: 3,
	},
	f x: 1 / (1 + x^2),
	g x: sin (2*x - 1),
},

plt.draw (\y: 2 * sqrt (1 - y^2 / 12)) p + \x: sin x / x = `
!            1            2   '~,       3            4            5             
!.4000-------+------------+-----',------+------------+------------+-------------
!            |            |      '~,    |            |            |             
!            |            |        ',   |            |            |             
!.2000-------+------------+---------',--+------------+------------+-------------
!            |            |          ', |            |            |             
!            |            |           ',|            |            |             
'~~~~,-------+------------+------------',------------+------------+-------------
$    '~~~~,  |            |             ',           |            |             
$         '~~,            |             |!           |            |             
0.8000-------'~~,---------+-------------+',----------+------------+-------------
$            |  '~,       |             | ',         |            |             
$            |    '~~,    |             |  !         |            |             
0.6000-------+-------'~,--+-------------+--',--------+------------+-------------
$            |         '~,|             |   !        |            |             
$            |           '~,            |   ',       |            |             
0.4000-------+------------+'~,----------+----!-------+------------+-------------
$            |            |  '~,        |    !       |            |             
$            |            |    '~,      |    !       |            |             
0.2000-------+------------+------'~,----+----'-------+------------+-------------
$            |            |        '~~, |            |            |             
$            |            |           '~,            |            |             
0.0000=======+============+============='~~,=========+============+=============
$            |            |             |  '~~,      |            |        ,~~~~
$            |            |             |     '~~~~, |            |  ,~~~~~'    
-0.199-------+------------+-------------+----------'~~~~~~~~~~~~~~~~~'----------
$            |            |             |            |            |             
$            |            |             |            |            |             
-0.399-------+------------+-------------+------------+------------+-------------
$            |            |             |            |            |             
`,
p: plt.Plot {
	ymin: -0.5, xcenter: 3, width: 6,
	rows: 30, cols: 80, divs: 4,
},

plt.Plot {
	width: 5, height: 5,
	rows: 30, cols: 80, labels: false,
} + [\x: 3/2 - 2*x, \x y: x^3 - 2 * x - y^2] = `
        |               |       ~,      $               |               | ,'    
        |               |        ',     $               |               |,'     
        |               |         ',    $               |               ,'      
--------+---------------+----------'~,--+---------------+-------------,~'-------
        |               |            ', $               |            ,' |       
        |               |             ',$               |           ,'  |       
        |               |              '~,              |          ,'   |       
        |               |               $',             |         ,'    |       
        |               ,~~~~~,         $ ',            |        ,'     |       
--------+-----------,~~~'-----'~~~,-----+--'~,----------+--------!------+-------
        |          ,'   |         '~,   $    ',         |       ,'      |       
        |         ,'    |           ',  $     ',        |      ,'       |       
        |        ,'     |            '~,$      '~,      |      !        |       
        |        !      |              !$        ',     |     ,'        |       
        |        !      |              !$         ',    |     !         |       
========+========!======+==============!#=========='~,==+=====!=========+=======
        |        !      |              !$            ', |     ',        |       
        |        ',     |            ,~'$             ',|      !        |       
        |         ',    |           ,'  $              '~,     ',       |       
        |          ',   |         ,~'   $               |',     ',      |       
        |           '~~~,     ,~~~'     $               | ',     !      |       
--------+---------------'~~~~~'---------+---------------+--'~,---',-----+-------
        |               |               $               |    ',   ',    |       
        |               |               $               |     ',   ',   |       
        |               |               $               |      '~,  ',  |       
        |               |               $               |        ',  ', |       
        |               |               $               |         ',  '~,       
--------+---------------+---------------+---------------+----------'~,--',------
        |               |               $               |            ', |',     
        |               |               $               |             '~| ',    
`,

